claimer/claimer/workers.go

```diff
@@ -79,89 +80,90 @@ func claimOnBinanceChain(bnbHTTP brpc.Client, claim server.ClaimJob) error {
 	return nil
 }
 
-func claimOnKava(config config.KavaConfig, client *KavaClient, claim server.ClaimJob, keyManager keys.KeyManager) error {
+func claimOnKava(config config.KavaConfig, client KavaChainClient, claim server.ClaimJob, privKey cryptotypes.PrivKey) error {
 	swapID, err := hex.DecodeString(claim.SwapID)
 	if err != nil {
 		return NewErrorFailed(err)
 	}
 
-	swap, err := client.GetAtomicSwap(swapID)
+	swap, err := client.GetSwapByID(swapID)
 	if err != nil {
 		return NewErrorRetryable(err)
 	}
-	if swap.Status == bep3.NULL {
+	if swap.Status == bep3types.SWAP_STATUS_UNSPECIFIED {
 		return NewErrorRetryable(fmt.Errorf("swap %s not found in state", swapID))
-	} else if swap.Status == bep3.Expired || swap.Status == bep3.Completed {
+	} else if swap.Status == bep3types.SWAP_STATUS_EXPIRED || swap.Status == bep3types.SWAP_STATUS_COMPLETED {
 		return NewErrorFailed(fmt.Errorf("swap %s has status %s and cannot be claimed", hex.EncodeToString(swapID), swap.Status))
 	}
 
-	fromAddr := keyManager.GetAddr()
+	fromAddr := sdk.AccAddress(privKey.PubKey().Address())
 
 	randomNumber, err := hex.DecodeString(claim.RandomNumber)
 	if err != nil {
 		return NewErrorFailed(err)
 	}
 
-	msg := bep3.NewMsgClaimAtomicSwap(fromAddr, swapID, randomNumber)
+	msg := bep3types.NewMsgClaimAtomicSwap(fromAddr.String(), swapID, randomNumber)
 	if err := msg.ValidateBasic(); err != nil {
 		return NewErrorFailed(fmt.Errorf("msg basic validation failed: \n%v", msg))
 	}
 
-	signMsg := &authtypes.StdSignMsg{
-		ChainID:       config.ChainID,
-		AccountNumber: 0,
-		Sequence:      0,
-		Fee:           calculateFee(ClaimTxDefaultGas, DefaultGasPrice),
-		Msgs:          []sdk.Msg{msg},
-		Memo:          "",
-	}
+	encodingConfig := client.GetEncodingCoding()
+
+	txBuilder := encodingConfig.TxConfig.NewTxBuilder()
+	txBuilder.SetMsgs(&msg)
+	txBuilder.SetGasLimit(ClaimTxDefaultGas)
+	txBuilder.SetFeeAmount(calculateFee(ClaimTxDefaultGas, DefaultGasPrice))
 
 	sequence, accountNumber, err := getAccountNumbers(client, fromAddr)
 	if err != nil {
 		return NewErrorFailed(err)
 	}
-	signMsg.Sequence = sequence
-	signMsg.AccountNumber = accountNumber
 
-	signedMsg, err := keyManager.Sign(*signMsg, client.cdc)
+	signerData := authsigning.SignerData{
+		ChainID:       config.ChainID,
+		AccountNumber: accountNumber,
+		Sequence:      sequence,
+	}
+
+	_, txBytes, err := signing.Sign(encodingConfig, privKey, txBuilder, signerData)
 	if err != nil {
-		return NewErrorFailed(err)
+		return err
 	}
-	tx := tmtypes.Tx(signedMsg)
 
-	maxTxLength := 1024 * 1024
-	if len(tx) > maxTxLength {
-		return NewErrorFailed(fmt.Errorf("the tx data exceeds max length %d ", maxTxLength))
+	request := txtypes.BroadcastTxRequest{
+		TxBytes: txBytes,
+		Mode:    txtypes.BroadcastMode_BROADCAST_MODE_SYNC,
```

Comment:
I think this could remove the need for the poll GetTxConfirmation below, not sure if this could be prone to errors though.  If keeping sync mode, bug below when getting tx confirmation

```suggestion
		Mode:    txtypes.BroadcastMode_BROADCAST_MODE_BLOCK,
```